04 Abstraction Exercises Part 1/delos-reyes-abstraction-exercise/src/main/java/problem/delosreyes/Main.java

```diff
@@ -0,0 +1,49 @@
+package main.java.problem.delosreyes;
+
+import java.util.Scanner;
+
+public class Main {
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        Rectangle newRectangle = new Rectangle();
+
+        int choice;
+        do {
+            System.out.println("Please choose what you want to do: ");
+            System.out.println("1. Enter unit of measurement.");
+            System.out.println("2. Compute for the area and perimeter of a rectangle.");
+            System.out.println("3. Quit.");
+            System.out.println("Enter your choice: ");
+            choice = sc.nextInt();
+            while (choice <= 0 || choice > 3) {
+                System.out.println("Choice must be between 1 and 3. Please enter a valid choice: ");
+                choice = sc.nextInt();
+            }
+
+            if (choice == 1) {
+                System.out.println("Enter unit of measurement: ");
+                String unit = sc.next();
+                newRectangle.setUnitOfMeasurement(unit);
```

Comment:
Again, remove the setter from Rectangle class. replace the setter with the following line:
`Rectangle.unitOfMeasurement = unit;`